### PR TITLE
Allow customisation of how all.js is loaded

### PR DESCRIPTION
**Pull request by tomsmalley**
If your page is mostly static, running the all.js blob immediately is not much of a concern.
When it does run, it will often cause the UI thread to stop for a significant chunk of time, meaning any animations which may be running at page load will be janky.
This change allows loading all.js at a later time, for example, after most of the animations have already finished.

I have:

  - [x] Based work on latest `develop` branch
  - [x] Looked for lint in my changes with `hlint .` (lint found code you did not write can be left alone)
  - [ ] Run the test suite: `$(nix-build -A selftest --no-out-link)`
  - [ ] (Optional) Run CI tests locally: `nix-build release.nix -A build.x86_64-linux --no-out-link` (or `x86_64-darwin` on macOS)

### CODE REVIEW

**Comment by tomsmalley** [comment]
This will certainly conflict with #644, but should be easy to resolve

File: lib/backend/src/Obelisk/Backend.hs [hunk 1]
```diff
@@ -193,16 +208,37 @@ runBackend backend frontend = case checkEncoder $ _backend_routeEncoder backend
 renderGhcjsFrontend
   :: (MonadSnap m, HasCookies m)
   => (route -> Text)
-  -> Text -- ^ URL to GHCJS app JavaScript
+  -> Text
+  -> (Text -> FrontendWidgetT route ()) -- ^ GHCJS all.js script widget
```
Comment (3noch) [inline lib/backend/src/Obelisk/Backend.hs]: Maybe it makes more sense to not pass both `allJsUrl` and this script everywhere. Why not "close over" the URL and pass just the script?

Comment (tomsmalley) [reply lib/backend/src/Obelisk/Backend.hs]: We use the path in the preload widget, which isn't customisable here. I thought about introducing a type to wrap this up, but didn't come to anything that felt "right".

Comment (3noch) [reply lib/backend/src/Obelisk/Backend.hs]: Fair.

Comment (tomsmalley) [reply lib/backend/src/Obelisk/Backend.hs]: I wonder if instead we should allow customising both preload widget and script widget? That shifts the problem slightly while giving the user a little more control.

Comment (tomsmalley) [reply lib/backend/src/Obelisk/Backend.hs]: The two widgets would fit fairly sensibly in a config record, I think.

Comment (3noch) [reply lib/backend/src/Obelisk/Backend.hs]: I like it.

File: lib/backend/src/Obelisk/Backend.hs [hunk 2]
```diff
@@ -58,19 +70,37 @@ data GhcjsApp route = GhcjsApp
   , _ghcjsApp_value :: !(Frontend route)
   }
 
+type GhcjsWidgets r = GhcjsWidgetsImpl (Text -> FrontendWidgetT r ())
+
+-- | Widgets used to load all.js on the frontend
+data GhcjsWidgetsImpl a = GhcjsWidgets
```
Comment (3noch) [inline lib/backend/src/Obelisk/Backend.hs]: Hm, I'm inclined to suggest that we drop the type alias and just make the ctor consistent name.

**Pull request merged**